frontend-test/unit/tabServiceSpec.js

```diff
@@ -96,11 +96,46 @@ define(['angular', 'angular-mocks'], function(angular) {
         var expectedResult = angular.copy(defaultExpectedResult);
         expectedResult.overview.active = true;
         expectedResult.preferences.enabled = false;
-        expectedResult.preferences.tooltip = 'Cannot elicit preferences because effects table contains missing values';
+        expectedResult.preferences.tooltip = 'Cannot elicit preferences because the effects table contains missing values.';
         expectedResult.deterministic.enabled = false;
-        expectedResult.deterministic.tooltip = 'Cannot perform analysis because effects table contains missing values';
+        expectedResult.deterministic.tooltip = 'Cannot perform analysis because the effects table contains missing values.';
         expectedResult.smaa.enabled = false;
-        expectedResult.smaa.tooltip = 'Cannot perform analysis because effects table contains missing values';
+        expectedResult.smaa.tooltip = 'Cannot perform analysis because the effects table contains missing values.';
+        expect(result).toEqual(expectedResult);
+      });
+
+      it('should disable the preferences, deterministic, and smaa tabs if there are more then 12 criteria in the effects table', function() {
+        workspaceServiceMock.checkForMissingValuesInPerformanceTable.and.returnValue(false);
+        const aggregateStateWithAlotOfCriteria = {
+          problem: {
+            criteria: {
```

Comment:
i would have done something like `_.fromPairs(_.map(_.range(0, 13), (n) => { return [n, {}]; }))` so that you can easily adjust the size and save some screen real estate